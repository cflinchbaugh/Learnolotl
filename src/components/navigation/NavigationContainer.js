import React, { Component } from 'react';

import { connect } from 'react-redux';
import { resetBuildData } from '../../actions/buildActions';

import Navigation from './Navigation';

export class NavigationContainer extends Component {
    constructor(props) {
        super(props);

        this.handleNavigationClick = this.handleNavigationClick.bind(this);
    }

    render() {
        let navigationData = {
                navigationItems: [
                    {
                        to: '/Learnolotl/learn',
                        onClick: this.handleNavigationClick,
                        label: 'Learn'
                    }, {
                        to: "/Learnolotl/build",
                        onClick: this.handleNavigationClick,
                        label: 'Build'
                    }, {
                        to: "/Learnolotl/about",
                        onClick: this.handleNavigationClick,
                        label: 'About'
                    }
                ]
            }
        
        return (
            <Navigation {...navigationData}/>
        );
    }

    handleNavigationClick() {
        this.props.resetBuildData();
    }
}


const mapStateToProps = state => ({

});

export default connect(mapStateToProps, { resetBuildData })(NavigationContainer);