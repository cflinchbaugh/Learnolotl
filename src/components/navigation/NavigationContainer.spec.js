import React from 'react';
import {shallow} from 'enzyme';
import {NavigationContainer} from './NavigationContainer';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('NavigationContainer', () => {
    it('renders and matches snapshot', () => {
        const renderer = new ShallowRenderer();

        const navigationContainerData = {

        }

        const tree = renderer.render(
            <NavigationContainer {...navigationContainerData}/>
        );

        expect(tree).toMatchSnapshot();
    });

    
});