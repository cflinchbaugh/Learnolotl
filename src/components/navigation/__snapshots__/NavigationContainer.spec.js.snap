// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationContainer renders and matches snapshot 1`] = `
<Navigation
  navigationItems={
    Array [
      Object {
        "label": "Learn",
        "onClick": [Function],
        "to": "/Learnolotl/learn",
      },
      Object {
        "label": "Build",
        "onClick": [Function],
        "to": "/Learnolotl/build",
      },
      Object {
        "label": "About",
        "onClick": [Function],
        "to": "/Learnolotl/about",
      },
    ]
  }
/>
`;
